src/cpp/shared_core/FilePath.cpp

```diff
@@ -1295,6 +1289,12 @@ void FilePath::setLastWriteTime(std::time_t in_time) const
    }
 }
 
+Error FilePath::testWritePermissions() const
+{
+   std::shared_ptr<std::ostream> testStream;
+   return openForWrite(testStream, false);
```

Comment:
Do you know if this could create an empty file on disk if the file doesn't exist? I'm not sure.

Reply:
Yes, it's very likely it will do this (good thought!)

Reply:
I updated the function so it no longer calls openForWrite, but borrows similar modified logic so that it only opens the file if it exists. I also updated the comment if FilePath.hpp to reflect this.